### PR TITLE
idris-modules: clarify usage of with-packages

### PR DESCRIPTION
**Pull request by samvher**
###### Motivation for this change

It took me a really long time (almost a day) to figure out how to use this, being unfamiliar with the way idris uses packages and with limited experience with nix expressions. The usage turned out to be rather simple but it would help if it's spelled out more. Also, if I actually include prelude and base in the desired libraries I get an error so I think that remark can be removed.
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/idris-modules/README.md [hunk 1]
```diff
@@ -34,6 +34,12 @@ Mostly for internal use.
...
 
-Bundle idris together with a list of packages. Because idris currently
-only supports a single directory in its library path, you must include
-all desired libraries here, including `prelude` and `base`.
\ No newline at end of file
+Bundle idris together with a list of packages. Example usage:
+
+    $ nix-shell -p "idrisPackages.with-packages [ idrisPackages.lightyear ]"
+
+Note that you still have to explicitly make the packages available to
```
Comment (FRidh) [inline pkgs/development/idris-modules/README.md]: try not to write "you"

**Comment by samvher** [comment]
I changed the text to not use "you".

**Comment by mpickering** [comment]
Looks good to me.

**Comment by Infinisil** [comment]
Needs to be updated, the docs have moved, and some stuff has changed

**Comment by Infinisil** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/50182

**Pull request closed without merging**